x-pack/plugins/maps/public/connected_components/mb_map/draw_control/draw_feature_control/draw_feature_control.tsx

```diff
@@ -57,12 +58,13 @@ export class DrawFeatureControl extends Component<Props, {}> {
             this.props.drawMode === DRAW_MODE.DRAW_POINTS
               ? feature.geometry.coordinates
               : feature.geometry;
-          this.props.addNewFeatureToIndex(
-            featureGeom,
-            this.props.drawShape || DRAW_SHAPE.SIMPLE_SELECT
-          );
+          featureGeometry.push(featureGeom);
         }
       });
+
+      if (Array.isArray(featureGeometry) && featureGeometry.length) {
```

Comment:
Why do you need Array.isArray check? featureGeometry is declared as an array. This check can be removed